### PR TITLE
feat: filter tabs

### PR DESCRIPTION
**Pull request by jake-bassett**

### CODE REVIEW

File: projects/assets-library/assets/styles/_layout.scss [hunk 1]
```diff
@@ -47,28 +47,33 @@
...
 }
 
+@mixin frame-border {
+  border: 1px solid $color-border;
+}
+
 @mixin card-hover {
   &:hover {
-    @include box-shadow();
+    @include dropdown-box-shadow();
```
Comment (anandtiwary) [inline projects/assets-library/assets/styles/_layout.scss:56]: Can we please move these to its own interaction.scss?

File: projects/components/src/filter-tabs/filter-tabs.component.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        [style.width.px]="this.activeElementPosition.width"
+      ></div>
+      <div class="container" *ngFor="let tab of this.tabs; let index = index">
+        <div class="divider" *ngIf="index !== 0" [class.adjacent-active]="this.isAdjacentActiveTab(index)"></div>
+        <htc-filter-tab class="tab" [label]="tab.label" (click)="this.setActiveTab(tab)"></htc-filter-tab>
+      </div>
+    </div>
+  `
+})
+export class FilterTabsComponent implements AfterViewInit, OnChanges {
```
Comment (anandtiwary) [inline projects/components/src/filter-tabs/filter-tabs.component.ts:37]: Is this going to replace toggle filter?

File: projects/components/src/filter-tabs/filter-tabs.component.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  public ngAfterViewInit(): void {
+    this.setActiveTab();
+  }
+
+  public setActiveTab(tab?: FilterTab): void {
+    this.activeTab = tab || (this.tabs && this.tabs[0]);
+    this.activeTabIndex = this.tabs?.indexOf(this.activeTab!);
+    this.activeElementPosition = this.getElementPosition(this.activeTab!, this.activeTabIndex);
+    this.activeTabChange.emit(this.activeTab);
+    this.changeDetector.markForCheck();
```
Comment (anandtiwary) [inline projects/components/src/filter-tabs/filter-tabs.component.ts:70]: Wouldn't this through View changed after check error?

Comment (anandtiwary) [reply projects/components/src/filter-tabs/filter-tabs.component.ts:70]: We could probably use QueryListAndChanges

File: projects/components/src/filter-tabs/filter-tabs.component.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  public isAdjacentActiveTab(index: number): boolean {
+    return (
+      this.activeTabIndex !== undefined && (index - this.activeTabIndex === 1 || index - this.activeTabIndex === 0)
```
Comment (anandtiwary) [inline projects/components/src/filter-tabs/filter-tabs.component.ts:75]: the rhs of second condition should also be compared against 1 isn't?

File: projects/components/src/filter-tabs/filter-tabs.component.ts (hunk 3 above)
Comment (aaron-steinfeld) [reply projects/components/src/filter-tabs/filter-tabs.component.ts:70]: Or just make the active tab an observable

File: projects/components/src/filter-tabs/filter-tabs.component.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  }
+
+  .active {
+    @include center-contents();
+    @include filter-box-shadow();
+    height: calc(100% - 4px); // Subtract 4px for margin
+    margin: 2px 0;
+    position: absolute;
+    width: 0; // Modified in html/ts
```
Comment (aaron-steinfeld) [inline projects/components/src/filter-tabs/filter-tabs.component.scss:34]: This is a simple tab, it does not seem from an outside look like a situation where we need to use absolute, measure, set explicit widths etc. What's preventing this from being done using more simple layout?

File: projects/assets-library/assets/styles/_layout.scss (hunk 1 above)
Comment (jake-bassett) [reply projects/assets-library/assets/styles/_layout.scss:56]: https://traceableai.atlassian.net/browse/ENG-5455

File: projects/components/src/filter-tabs/filter-tabs.component.ts (hunk 2 above)
Comment (jake-bassett) [reply projects/components/src/filter-tabs/filter-tabs.component.ts:37]: Toggle Button? No, this component is specifically for toggle filters. The toggle buttons will still exist but won't be used for filters anymore.

File: projects/components/src/filter-tabs/filter-tabs.component.ts (hunk 4 above)
Comment (jake-bassett) [reply projects/components/src/filter-tabs/filter-tabs.component.ts:75]: Not sure what you mean. What we want to do here is remove the divider lines surrounding the active filter so we compare our index to the active one. If the diff is 0 or 1 we know we are next to the active tab.

File: projects/components/src/filter-tabs/filter-tabs.component.scss (hunk 5 above)
Comment (jake-bassett) [reply projects/components/src/filter-tabs/filter-tabs.component.scss:34]: This is all to support the animations. The animation requires the movement of an element. But we only want the white background to move back and forth (left position) and adjust its width to match the parent tab.

**Comment by jake-bassett** [comment]
> My high level feedback - what is this component's purpose compared to the toggle button? This doesn't seem at all related to filters, even if that's one use case for it. I don't think we need both things, so maybe whatever functionality or styling that is missing from toggle can be added?

This component is _only_ for filters. The toggle buttons are still a thing, but they will not be used for filters anymore.

File: projects/components/src/filter-tabs/filter-tabs.component.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      </div>
+    </div>
+  `
+})
+export class FilterTabsComponent implements AfterViewInit, OnChanges {
+  @Input()
+  public readonly tabs?: FilterTab[] = [];
+
+  @Input()
+  public activeTab?: FilterTab;
```
Comment (jake-bassett) [inline projects/components/src/filter-tabs/filter-tabs.component.ts:41]: This should be readonly. Fix incoming.

File: projects/components/src/filter-tabs/filter-tabs.component.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      this.setActiveTab();
+    }
+  }
+
+  public ngAfterViewInit(): void {
+    this.setActiveTab();
+  }
+
+  public setActiveTab(tab?: FilterTab): void {
+    this.activeTab = tab || (this.tabs && this.tabs[0]);
```
Comment (jake-bassett) [inline projects/components/src/filter-tabs/filter-tabs.component.ts:63]: Should not reassign an Input. Fix incoming.

**Pull request closed without merging**